x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/new_terms/utils.ts

```diff
@@ -46,3 +50,94 @@ export const validateHistoryWindowStart = ({
     );
   }
 };
+
+/**
+ * Takes a list of buckets and creates value from them to be used in 'include' clause of terms aggregation.
+ * For a single new terms field, value equals to bucket name
+ * For multiple new terms fields and buckets, value equals to concatenated base64 encoded bucket names
+ * @returns for buckets('host-0', 'test'), resulted value equals to: 'aG9zdC0w_dGVzdA=='
+ */
+export const transformBucketsToValues = (
+  newTermsFields: string[],
+  buckets: estypes.AggregationsCompositeBucket[]
+): Array<string | number> => {
+  // if new terms include only one field we don't use runtime mappings and don't stich fields buckets together
+  if (newTermsFields.length === 1) {
+    return buckets
+      .map((bucket) => Object.values(bucket.key)[0])
+      .filter((value): value is string | number => value != null);
+  }
+
+  return buckets.map((bucket) =>
+    Object.values(bucket.key)
+      .filter((value): value is string | number => value != null)
```

Comment:
If we filter out individual `null` values from a particular bucket, is there any chance of the encoded value from the bucket matching the runtime field-calculated value since the runtime field will still use all of the fields, including the nulls? I wonder if we should exclude the entire bucket if any values are `null`, or include the nulls and base64 encode them as well.

Reply:
Looks like, we don't need this check at all.
Our query is following
```
POST /null-*/_search?allow_no_indices=true&ignore_unavailable=true
{
   "query":{
      "bool":{
         "filter":[
            {
               "bool":{
                  "must":[
                     
                  ],
                  "filter":[
                     {
                        "query_string":{
                           "query":"*"
                        }
                     }
                  ],
                  "should":[
                     
                  ],
                  "must_not":[
                     
                  ]
               }
            },
            {
               "range":{
                  "@timestamp":{
                     "lte":"2022-11-10T14:46:11.680Z",
                     "gte":"2022-11-08T14:45:11.680Z",
                     "format":"strict_date_optional_time"
                  }
               }
            },
            {
               "match_all":{
                  
               }
            }
         ]
      }
   },
   "fields":[
      {
         "field":"*",
         "include_unmapped":true
      },
      {
         "field":"@timestamp",
         "format":"strict_date_optional_time"
      }
   ],
   "aggregations":{
      "new_terms":{
         "composite":{
            "sources":[
               {
                  "host.name":{
                     "terms":{
                        "field":"host.name"
                     }
                  }
               }
            ],
            "size":10000
         }
      }
   },
   "runtime_mappings":{
      
   },
   "sort":[
      {
         "@timestamp":{
            "order":"asc",
            "unmapped_type":"date"
         }
      }
   ],
   "size":0
}
```

And it doesn't return null values in buckets. Per [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-composite-aggregation.html#_missing_bucket), we need to introduce a parameter in               `"missing_bucket": true,` on sources terms item.
I think we can dismiss this check entirely then. Downside of this, we would need to cast typing here, as bucket type here is following:
```ts
export interface AggregationsCompositeBucketKeys extends AggregationsMultiBucketBase {
    key: Record<string, any>;
}
```

On the other note: should we treat null value in term as a new one? In current implementation we dismiss nulls. But could it be a valid case: some field has become nullable for the first time in comparison to historical,  do we want to generate alert about it?

Reply:
added filtering for buckets with multiple fields and functional tests for it